lib/controller.py

```diff
@@ -453,6 +459,8 @@ def get_root_url(self):
             action_module_path = action_module_path[1:]
         if 'HTTP_X_FORWARDED_PROTO' in self.environ:
             protocol = self.environ['HTTP_X_FORWARDED_PROTO']
+        elif 'HTTP_X_FORWARDED_PROTOCOL' in self.environ:
+            protocol = self.environ['HTTP_X_FORWARDED_PROTOCOL']
```

Comment:
What do we need this for?

Reply:
after successful authn. in loginx you are redirected to
```
self._redirect('%sfirst_form' % (self.get_root_url(), ))
```
without the above, you would get to http (even if you have https proxy from lindat)